### PR TITLE
Add missing import to compile micro kernels on Windows.

### PR DESCRIPTION
**Pull request by AdamHillier**
This inclusion allows building the TFLite micro kernels on Windows.

Specifically, `bazel build -c opt //tensorflow/lite/micro/kernels:micro_ops` currently throws a syntax error without this PR, on this line: https://github.com/tensorflow/tensorflow/blob/f7d0a77b539d47bbbffa9702c1e7329de79551ac/tensorflow/lite/micro/kernels/svdf.cc#L55-L60

This fixes the issue by including the correct definition of `__restrict__` for MSVC: https://github.com/tensorflow/tensorflow/blob/f7d0a77b539d47bbbffa9702c1e7329de79551ac/tensorflow/lite/kernels/internal/tensor_utils.h#L24-L26

### CODE REVIEW

**Review by njeffrie — APPROVED** [review]

File: tensorflow/lite/micro/kernels/svdf.cc [hunk 1]
```diff
@@ -22,6 +22,7 @@ limitations under the License.
 #include "tensorflow/lite/kernels/internal/common.h"
 #include "tensorflow/lite/kernels/internal/quantization_util.h"
 #include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/internal/tensor_utils.h"
```
Comment (njeffrie) [inline tensorflow/lite/micro/kernels/svdf.cc:25]: Looks to me like this is bringing in some non-TFLM compatible dependencies. Perhaps try using ```tensorflow/lite/kernels/internal/reference/portable_tensor_utils.h```

**Review by njeffrie — APPROVED** [review]

File: tensorflow/lite/micro/kernels/svdf.cc (hunk 1 above)
Comment (AdamHillier) [reply tensorflow/lite/micro/kernels/svdf.cc:25]: Ah thanks, yes will try that :+1:

Comment (AdamHillier) [reply tensorflow/lite/micro/kernels/svdf.cc]: I've updated the PR in 9c8f7f7. Is there a way that I can run the TFLM tests locally to test small changes like this?

Comment (njeffrie) [reply tensorflow/lite/micro/kernels/svdf.cc]: It should be possible to run ```make -f tensorflow/lite/micro/tools/make/Makefile test``` to catch some issues like this.

Sometimes issues pop up on certain targets and not others. In this case, I tend to look at the test log to see the last make or bazel command which caused the error. Running that command locally should reproduce the same error most of the time.

**Review by njeffrie — APPROVED** [review]

**Comment by njeffrie** [comment]
Please run ```clang-format -i -style=google tensorflow/lite/micro/kernels/svdf.cc```

**Comment by AdamHillier** [comment]
Thanks for your help with this - after the linting, I ran `make -f tensorflow/lite/micro/tools/make/Makefile test` locally as you suggested and got a bunch of undefined-symbol errors, which I guess I should have expected. Since the thing I actually want is only three lines long, would it be a problem if I just copy those three lines to the top of the `svdf.cc` file instead of adding this extra import?

https://github.com/tensorflow/tensorflow/blob/66dd98a10612e644a26a141cbdfc82e55ecf62eb/tensorflow/lite/kernels/internal/reference/portable_tensor_utils.h#L20-L22

**Comment by njeffrie** [comment]
I tried running make ... test with your change and I'm not seeing any errors. Make sure you run make ... third_party_downloads first to ensure you have all dependencies downloaded correctly - that my fix your issue.

**Comment by AdamHillier** [comment]
Apologies for the delay, I just pushed the linted file, so we can see what happens on CI.

What I see (on MacOS) when running the tests is a bunch of `undefined symbol` errors because `portable_tensor_utils.h` is being included but `portable_tensor_utils.cc` isn't being compiled. However, when I added `portable_tensor_utils.cc` to the `MICROLITE_CC_BASE_SRCS` in the makefile I then got a bunch of errors from that file because of `-Wdouble-promotion`, so that doesn't (immediately) solve the problem either.

**Review by njeffrie — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by advaitjain** [comment]
I'm going to close the current PR. If this is still an issue please feel free to create a new PR that removes `__restrict__` since that is an optimization which is not necessary in the reference implementation. The benefits of `__restrict__` in svdf are not entirely clear either so removing it to fix the Windows build will be ok.

**Pull request closed without merging**